### PR TITLE
Adds pre-commit checks to the repo. -- SGPROJ-1567

### PR DESCRIPTION
**Pull request by DJKaswa**
## Change description

> Please review this and feel free to fix anything that will block it from passing on Github; otherwise it should be good to go as it passes all checks locally.

> Once you review and approve it, you can merge it to main (don’t forget to delete the branch). At that point, it’ll be ready to be moved to tf-auto.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -26,42 +26,43 @@ PATTERN
...
 data "archive_file" "function_code" {
-  type = "zip"
-  source_dir = "${path.module}/function_code"
+  type        = "zip"
+  source_dir  = "${path.module}/function_code"
   output_path = "${path.module}/function_code_zipped/function_code.zip"
 }
 
 #Function to process event
+#tfsec:ignore:aws-lambda-enable-tracing -- Ignore warning on Function not having tracing enabled.
```
Comment (JoseMotto-sg) [inline config.tf]: Please add log for lambda
```suggestion
```

File: config.tf [hunk 2]
```diff
@@ -94,11 +95,12 @@ EOF
 }
 
 #Default policy for Lambda to be executed and put logs in Cloudwatch
+#tfsec:ignore:aws-iam-no-policy-wildcards -- Ignores warning on wildcarded resource
```
Comment (JoseMotto-sg) [inline config.tf]: Please point to the granular resource instead of wildcard the policy
```suggestion
```

File: config.tf [hunk 3]
```diff
@@ -128,11 +130,12 @@ EOF
 }
 
 #Policy for additional Permissions for Lambda Execution
+#tfsec:ignore:aws-iam-no-policy-wildcards -- Ignores warning on wildcarded resource
```
Comment (JoseMotto-sg) [inline config.tf]: Please point to the granular resource instead of wildcard the policy
```suggestion
```

File: config.tf [hunk 4]
```diff
@@ -159,11 +162,11 @@ EOF
 }
 
 #Cloudwatch Log Group for Function
+#tfsec:ignore:aws-cloudwatch-log-group-customer-key --Ignore warning on Log group not being encrypted.
```
Comment (JoseMotto-sg) [inline config.tf]: Please create a KMS key for the resource
```suggestion
```

**Review by JoseMotto-sg — CHANGES_REQUESTED** [review]

**Comment by DJKaswa** [comment]
@JoseMotto-sg please review again.

File: config.tf [hunk 5]
```diff
@@ -159,11 +167,19 @@ EOF
 }
 
 #Cloudwatch Log Group for Function
+#tfsec:ignore:aws-cloudwatch-log-group-customer-key -- Ignore warning on Log group not being encrypted.
```
Comment (JoseMotto-sg) [inline config.tf]: Please Add KMS key for the resource
```suggestion
```

**Review by JoseMotto-sg — CHANGES_REQUESTED** [review]

**Review by JoseMotto-sg — APPROVED** [review]
LGTM

**Pull request merged**